Shaders/ssr_pass/ssr_pass.frag.glsl

```diff
@@ -92,14 +92,14 @@ void main() {
 	n = normalize(n);
 
 	vec3 viewNormal = V3 * n;
-	vec3 viewPos = getPosView(viewRay, d, cameraProj);
+	vec3 viewPos = normalize(getPosView(viewRay, d, cameraProj));
```

Comment:
Regarding all the `normalize()` changes from https://github.com/armory3d/armory/pull/2652/commits/f5e85a500c0485bec953fd2be63e6d7aa315e5de, how does it happen that the value isn't normalized in the first place? Shouldn't a reflection of the already normalized `viewPos` be normalized again? Was the normal vector not normalized or were there rounding issues that could result in values slightly larger than 1?

Reply:
Oops, sorry, this is the wrong location for this comment. I am referring to the other `normalize()` calls introduced in https://github.com/armory3d/armory/pull/2652/commits/f5e85a500c0485bec953fd2be63e6d7aa315e5de

Reply:
Yes I don't get it, yet this is the only solution for refraction, so I applied it to reflection too.

Reply:
All right, then I assume it's due to floating point errors. For things like these a graphics debugger like RenderDoc is really helpful btw :)